src/lib/server/websearch/markdown/fromHtml.ts

```diff
@@ -0,0 +1,98 @@
+import { collapseString, sanitizeString } from "./utils/nlp";
+import { stringifyHTMLElements, stringifyHTMLElementsUnformatted } from "./utils/stringify";
+import { MarkdownElementType, tagNameMap, type HeaderElement, type MarkdownElement } from "./types";
+import type { SerializedHTMLElement } from "../scrape/types";
+
+interface ConversionState {
+	defaultType:
+		| MarkdownElementType.Paragraph
+		| MarkdownElementType.BlockQuote
+		| MarkdownElementType.UnorderedListItem
+		| MarkdownElementType.OrderedListItem;
+	listDepth: number;
+	blockQuoteDepth: number;
```

Comment:
```suggestion
	depth?: number; // to be used for list depth or blockquote depth
```

wdyr about this typing?

Reply:
It's possible to have lists inside of block quotes so it's good to keep these separate